### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by cammellos**
After our discussion, I have updated our CONTRIBUTING.md. I have removed the starter resources as they are quite old and likely stale.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,37 +1,74 @@
...
+   - [Metrics for Test Coverage](#metrics-for-test-coverage)
 
 ## Workflow
 
 1. Pick an [issue](https://github.com/status-im/status-go/issues) to work on and drop a line there that you're working on that.
-2. Wait for an approve from one of core contributors.
+2. Wait for approval from one of the core contributors.
 3. Fork the project and work right in the `develop` branch.
 4. Work on the issue and file a PR back into `develop`.
 5. Wait until your PR is [reviewed](https://hackmd.io/s/B1AenvFU-) by 2 core developers and merged.
```
Comment (churik) [inline CONTRIBUTING.md:25]: seems that 5 in particular we should force on GH level - to prevent PR from merge unless it has 2 reviews

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,37 +1,74 @@
...
+## Testing
+
+### Test Validation
+
+Every Pull Request (PR) should include tests to validate its correctness and to test the features implemented. Preferably, use Behavior-Driven Development (BDD) principles for writing tests.
+
+### Area of Impact
+
+PRs should be well described, and the description should clearly specify the area of impact. This helps reviewers and maintainers understand the changes being made.
+
```
Comment (churik) [inline CONTRIBUTING.md:55]: I'd add a line about affecting both clients (desktop/mobile) in an observable way and request manual testing if needed from dev POV.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,37 +1,74 @@
...
+1. **Squash PR before merging**: You can do it either with GitHub API by merging with "Squash and merge" or locally if you want to preserve your signature. It is ok to merge multiple commits with "Rebase and merge" if they are logically separate.
+
+2. **Preserve as much history as possible**: If you need to re-name a file, use `git mv` - it will preserve git history.
+
+## Testing
+
+### Test Validation
+
+Every Pull Request (PR) should include tests to validate its correctness and to test the features implemented. Preferably, use Behavior-Driven Development (BDD) principles for writing tests.
```
Comment (J-Son89) [inline CONTRIBUTING.md]: while BDD is fairly common perhaps there's a useful link for this?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,37 +1,74 @@
...
+
+### Area of Impact
+
+PRs should be well described, and the description should clearly specify the area of impact. This helps reviewers and maintainers understand the changes being made.
+
+## Pull Request Description
+
+### Feature Flags
+
+For PRs introducing new features, especially those in high-risk areas, consider using feature flags. Feature flags allow features to be toggled on or off, providing a way to deploy code changes to production while controlling the visibility of new features.
```
Comment (J-Son89) [inline CONTRIBUTING.md:60]: Imo Feature flags is a nice idea, but can often be tricky to implement correctly. It would be nice if there was some good clear examples to show these being added/ a clean approach to do so in this codebase in particular

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,37 +1,74 @@
...
+
+1. Before submitting a PR, check the existing test coverage for the modified or new code.
+2. Ensure that the new code is covered by relevant unit tests, integration tests, or other appropriate testing methods.
+3. If modifying existing code, update or add tests to cover the changes.
+4. If adding new features, include tests that demonstrate the correct functionality and handle edge cases.
+
+### Metrics for Test Coverage
 
-## Keep history clean
+PRs should not decrease the overall coverage. If possible, aim to increase the overall coverage with each contribution.
```
Comment (J-Son89) [inline CONTRIBUTING.md:80]: we should be able to automate this, no?

**Comment by J-Son89** [comment]
One minor detail and something we do in the mobile repo which helps keeps things organised is to assign the right people once a pr has been opened. Can we start adding this step to Go prs? 
I even just use this to see my own pr's and ensure I am never opening too many at a time.

**Comment by J-Son89** [comment]
Also I see there is 71 open pr's. We should be closing stale/old pr's. 
We're fortunate enough to have @flexsurfer do this for Status-mobile. 🙌 
Would be nice if someone could stay on top of this in this repo! 👍

File: CONTRIBUTING.md (hunk 4 above)
Comment (cammellos) [reply CONTRIBUTING.md:60]: I can add a link, we used them in messenger quite a few times https://github.com/status-im/status-go/blob/develop/protocol/common/feature_flags.go

File: CONTRIBUTING.md (hunk 5 above)
Comment (cammellos) [reply CONTRIBUTING.md:80]: Yes, that we could I think

File: CONTRIBUTING.md (hunk 1 above)
Comment (cammellos) [reply CONTRIBUTING.md:25]: Once there's agreement, I can set it up

File: CONTRIBUTING.md (hunk 3 above)
Comment (cammellos) [reply CONTRIBUTING.md]: updated

File: CONTRIBUTING.md (hunk 2 above)
Comment (cammellos) [reply CONTRIBUTING.md:55]: added

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,37 +1,76 @@
...
 
 ## Code Style
 
-Please, note that we follow [Effective Go](https://golang.org/doc/effective_go.html) and
-[CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments) in our code.
+Please note that we follow [Effective Go](https://golang.org/doc/effective_go.html) and [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments) in our code.
+
+## Keep History Clean
+
+1. **Squash PR before merging**: You can do it either with GitHub API by merging with "Squash and merge" or locally if you want to preserve your signature. It is ok to merge multiple commits with "Rebase and merge" if they are logically separate.
```
Comment (J-Son89) [inline CONTRIBUTING.md:33]: we have this mostly blocked on mobile repo, is it possible to do in this repo too? seems preferable to automate.

Comment (cammellos) [reply CONTRIBUTING.md:33]: you mean to prevent merging from the UI?

Comment (J-Son89) [reply CONTRIBUTING.md:33]: I mean that commits are squashed.
On mobile repo it will squash for you when you go to merge

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,37 +1,76 @@
...
+
+2. **Preserve as much history as possible**: If you need to re-name a file, use `git mv` - it will preserve git history.
+
+## Testing
+
+### Test Validation
+
+Every Pull Request (PR) should include tests to validate its correctness and to test the features implemented. Preferably, use [Behavior-Driven Development (BDD)](https://en.wikipedia.org/wiki/Behavior-driven_development) principles for writing tests.
```
Comment (igor-sirotin) [inline CONTRIBUTING.md:49]:
```suggestion
Every Pull Request (PR) should include tests to validate its correctness and to test the features implemented or the bug fixed. Preferably, use [Behavior-Driven Development (BDD)](https://en.wikipedia.org/wiki/Behavior-driven_development) principles for writing tests.
```

File: CONTRIBUTING.md (hunk 5 above)
Comment (igor-sirotin) [reply CONTRIBUTING.md:80]: I noted this already, but just in case:
It should be possible to set this up with codeclimate. It's just not configured for our project:
https://codeclimate.com/github/status-im/status-go

image

**Review by igor-sirotin — APPROVED** [review]
❤️

**Comment by igor-sirotin** [comment]
image

lol ios build couldn't make these changes 😄
